yktrace/src/sir.rs

```diff
@@ -33,7 +34,11 @@ pub struct Sir<'m> {
     /// The current executable's ELF information.
     exe_obj: object::File<'m>,
     /// Section cache to avoid expensive `object::File::section_by_name()` calls.
-    sec_cache: FxHashMap<String, &'m [u8]>
+    sec_cache: FxHashMap<String, &'m [u8]>,
+    /// Body cache, to avoid repeated decodings.
+    body_cache: RwLock<FxHashMap<String, Option<Arc<Body>>>>,
```

Comment:
What if instead of having a lock around the map, you have a map that is pre-filled with a lock for each body? While this requires decoding the body names, you only need to decode the actual body when it is used. This could reduce contention a lot.

Reply:
I think that's similar to what Laurie proposed above.

I'm going to try this (and a few other things) tomorrow.

Reply:
Missed that comment.